diff --git a/common-src/Makefile.am b/common-src/Makefile.am
index 40e2eb6..93520b5 100644
--- a/common-src/Makefile.am
+++ b/common-src/Makefile.am
@@ -146,6 +146,7 @@ genversion_SOURCES = genversion.c svn-info.h
 genversion_LDADD = $(libamanda_la_LIBADD)	\
 	../gnulib/libgnu.la
 
+genversion.@OBJEXT@: $(genversion_SOURCES) genversion.h
 genversion.h: $(top_builddir)/config.status
 	-rm -f $@ $@.new
 	echo '#define CC "$(CC)"' > $@.new
